src/progress_bar.rs

```diff
@@ -55,8 +55,16 @@ impl ProgressBar {
     }
 
     /// A convenience builder-like function for a progress bar with a given style
-    pub fn with_style(self, style: ProgressStyle) -> ProgressBar {
-        self.state().style = style;
+    pub fn with_style(self, mut style: ProgressStyle) -> ProgressBar {
+        {
+            let mut state = self.state();
+            // Retain current message and prefix - they are actually stored as part of the style,
+            // however the user probably doesn't know this.
+            style.message = state.style.message.clone();
```

Comment:
We can probably do this without cloning. Maybe `mem::swap()` and then overwrite?

Style nit: I'd prefer to just call `drop(state);` instead of having the nested scope.

Reply:
Will do!